src/platform/packages/shared/react/kibana_context/root/eui_provider.tsx

```diff
@@ -19,15 +20,22 @@ import {
   getThemeConfigByName,
   DEFAULT_THEME_CONFIG,
 } from '@kbn/react-kibana-context-common';
-import type { UserProfileService } from '@kbn/core-user-profile-browser';
 import type { ThemeServiceStart } from '@kbn/react-kibana-context-common';
 
+interface IUserProfile {
+  getUserProfile$: () => Rx.Observable<Record<string, unknown> | null>;
+}
+
+interface UserSettings {
+  contrastMode: 'system' | 'standard' | 'high';
+}
+
 /**
  * Props for the KibanaEuiProvider.
  */
 export interface KibanaEuiProviderProps extends Pick<EuiProviderProps<{}>, 'modify' | 'colorMode'> {
   theme: ThemeServiceStart;
-  userProfile?: Pick<UserProfileService, 'getUserProfile$'>; // TODO: use this to access a "high contrast mode" flag from user settings. Pass the flag to EuiProvider, when it is supported in EUI.
+  userProfile?: IUserProfile;
```

Comment:
nit: tbh, I preferred the previous version 😅 not sure if there was a good reason for the change